ibm/service/logsrouting/resource_ibm_logs-router_tenant.go

```diff
@@ -0,0 +1,823 @@
+	model := &ibmcloudlogsroutingv0.TargetTypePrototypeTargetTypeLogsPrototype{}
+	model.LogSinkCRN = core.StringPtr(modelMap["log_sink_crn"].(string))
+	model.Name = core.StringPtr(modelMap["name"].(string))
+	if modelMap["parameters"] != nil && len(modelMap["parameters"].([]interface{})) > 0 {
+		ParametersModel, err := ResourceIBMLogsRouterTenantMapToTargetParametersTypeLogsPrototype(modelMap["parameters"].([]interface{})[0].(map[string]interface{}))
+		if err != nil {
+			return model, err
+		}
+		model.Parameters = ParametersModel
+	}
+	return model, nil
+}
+
+func ResourceIBMLogsRouterTenantMapToTargetParametersTypeLogsPrototype(modelMap map[string]interface{}) (*ibmcloudlogsroutingv0.TargetParametersTypeLogsPrototype, error) {
+	model := &ibmcloudlogsroutingv0.TargetParametersTypeLogsPrototype{}
+	model.Host = core.StringPtr(modelMap["host"].(string))
+	model.Port = core.Int64Ptr(int64(modelMap["port"].(int)))
+	return model, nil
+}
+
+func ResourceIBMLogsRouterTenantTargetTypeToMap(model ibmcloudlogsroutingv0.TargetTypeIntf) (map[string]interface{}, error) {
+	if _, ok := model.(*ibmcloudlogsroutingv0.TargetTypeLogDna); ok {
+		return ResourceIBMLogsRouterTenantTargetTypeLogDnaToMap(model.(*ibmcloudlogsroutingv0.TargetTypeLogDna))
+	} else if _, ok := model.(*ibmcloudlogsroutingv0.TargetTypeLogs); ok {
+		return ResourceIBMLogsRouterTenantTargetTypeLogsToMap(model.(*ibmcloudlogsroutingv0.TargetTypeLogs))
+	} else if _, ok := model.(*ibmcloudlogsroutingv0.TargetType); ok {
+		modelMap := make(map[string]interface{})
+		model := model.(*ibmcloudlogsroutingv0.TargetType)
+		if model.ID != nil {
+			modelMap["id"] = model.ID.String()
+		}
+		if model.LogSinkCRN != nil {
+			modelMap["log_sink_crn"] = *model.LogSinkCRN
+		}
+		if model.Name != nil {
+			modelMap["name"] = *model.Name
+		}
+		if model.Etag != nil {
+			modelMap["etag"] = *model.Etag
+		}
+		if model.Type != nil {
+			modelMap["type"] = *model.Type
+		}
+		if model.CreatedAt != nil {
+			modelMap["created_at"] = *model.CreatedAt
+		}
+		if model.UpdatedAt != nil {
+			modelMap["updated_at"] = *model.UpdatedAt
+		}
+		if model.Parameters != nil {
+			parametersMap, err := ResourceIBMLogsRouterTenantTargetParametersTypeLogDnaToMap(model.Parameters)
+			if err != nil {
+				return modelMap, err
+			}
+			modelMap["parameters"] = []map[string]interface{}{parametersMap}
+		}
+		return modelMap, nil
+	} else {
+		return nil, fmt.Errorf("Unrecognized ibmcloudlogsroutingv0.TargetTypeIntf subtype encountered")
+	}
+}
+
+func ResourceIBMLogsRouterTenantTargetParametersTypeLogDnaToMap(model *ibmcloudlogsroutingv0.TargetParametersTypeLogDna) (map[string]interface{}, error) {
+	modelMap := make(map[string]interface{})
+	modelMap["host"] = *model.Host
+	modelMap["port"] = flex.IntValue(model.Port)
+	modelMap["access_credential"] = REDACTED_TEXT // pragma: whitelist secret
```

Comment:
The access_credential is a secret value, and I added this in the printout functions to obfuscate the value. This is not necessary because that parameter is set as sensitive, so it will never be printed out by terraform
```
"access_credential": &schema.Schema{
	Type:        schema.TypeString,
	Optional:    true,
	Sensitive:   true,
	Description: "Secret to connect to the log-sink",
},
```
All occurrences of `REDACTED_TEXT` have been removed